diff --git a/src/IncrementalBundler.js b/src/IncrementalBundler.js
index d08a90fa6f12816cb799746b493d438074b6df65..3ec78ba7b039965e9f14496fae0ce3089ee71acd 100644
--- a/src/IncrementalBundler.js
+++ b/src/IncrementalBundler.js
@@ -70,8 +70,7 @@ class IncrementalBundler {
       lazy: otherOptions.lazy,
       unstable_allowRequireContext:
         this._config.transformer.unstable_allowRequireContext,
-      unstable_enablePackageExports:
-        this._config.resolver.unstable_enablePackageExports,
+      unstable_enablePackageExports: true,
       shallow: otherOptions.shallow,
     });
     this._config.serializer.experimentalSerializerHook(graph, {
@@ -114,8 +113,7 @@ class IncrementalBundler {
         lazy: otherOptions.lazy,
         unstable_allowRequireContext:
           this._config.transformer.unstable_allowRequireContext,
-        unstable_enablePackageExports:
-          this._config.resolver.unstable_enablePackageExports,
+        unstable_enablePackageExports: true,
         shallow: otherOptions.shallow,
       }
     );
diff --git a/src/lib/getPrependedScripts.js b/src/lib/getPrependedScripts.js
index 8e168ce6a5c3d4e66cd99a253a6d416f1546dd32..5f4fdfba63edee02e451c82eaa6c3631f71c4662 100644
--- a/src/lib/getPrependedScripts.js
+++ b/src/lib/getPrependedScripts.js
@@ -45,8 +45,7 @@ async function getPrependedScripts(
       transformOptions,
       onProgress: null,
       lazy: false,
-      unstable_enablePackageExports:
-        config.resolver.unstable_enablePackageExports,
+      unstable_enablePackageExports: true,
       shallow: false,
     }
   );
diff --git a/src/lib/transformHelpers.js b/src/lib/transformHelpers.js
index b6f7897fb88b3bc406c75d50585a6fc2f94565b8..36579488f49741673dd8932aff4405a92c55fd9e 100644
--- a/src/lib/transformHelpers.js
+++ b/src/lib/transformHelpers.js
@@ -62,7 +62,7 @@ async function calcTransformerOptions(
       unstable_allowRequireContext:
         config.transformer.unstable_allowRequireContext,
       unstable_enablePackageExports:
-        config.resolver.unstable_enablePackageExports,
+        true,
       shallow: false,
     });
     return Array.from(dependencies.keys());
diff --git a/src/node-haste/DependencyGraph.js b/src/node-haste/DependencyGraph.js
index 7fda8516588a691e4fc1a81dcd38b72e715d95ec..613b8468b704e75b660cbaf652023e89581a81b8 100644
--- a/src/node-haste/DependencyGraph.js
+++ b/src/node-haste/DependencyGraph.js
@@ -155,7 +155,7 @@ class DependencyGraph extends EventEmitter {
       unstable_conditionsByPlatform:
         this._config.resolver.unstable_conditionsByPlatform,
       unstable_enablePackageExports:
-        this._config.resolver.unstable_enablePackageExports,
+        true,
       unstable_getRealPath: this._config.resolver.unstable_enableSymlinks
         ? getRealPathIfFile
         : null,
